From 9bfa7f45e2dd678e314ed98bfac7e0f4d2d4d765 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Fri, 28 Jun 2024 18:21:39 +0300
Subject: [PATCH] [FSW-420] fs1: initialize key revocation bits

with this patch, added support to initialize
security data structure with key revocation bits
and added function to check revoke status.

Change-Id: I8adae55c0438acb62868254900d773e8b853997d
---

diff --git a/arch/fs1/arc/Makefile b/arch/fs1/arc/Makefile
index ddb3593..ecb3e98 100644
--- a/arch/fs1/arc/Makefile
+++ b/arch/fs1/arc/Makefile
@@ -14,3 +14,4 @@
 BOOT_ROM_COMMON_SRCS_C += $(shell ls $(HL_DEV_ARCH)/dev_recovery.c)
 BOOT_ROM_COMMON_SRCS_C += $(shell ls $(HL_DEV_ARCH)/dev_boot_info.c)
 BOOT_ROM_COMMON_SRCS_C += $(shell ls $(HL_DEV_ARCH)/dev_dma.c)
+BOOT_ROM_COMMON_SRCS_C += $(shell ls $(HL_DEV_ARCH)/dev_security.c)
diff --git a/arch/fs1/arc/dev_security.c b/arch/fs1/arc/dev_security.c
new file mode 100644
index 0000000..58b73d6
--- /dev/null
+++ b/arch/fs1/arc/dev_security.c
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier:     GPL-2.0+
+/*
+ * Copyright (C) 2024 HabanaLabs Ltd.
+ * All Rights Reserved.
+ */
+
+#include <boot_rom.h>
+#include <mem_io.h>
+#include <fs1_blocks_linux_driver.h>
+#include <psoc_security_regs.h>
+#include <fs1_reg_map.h>
+#include <fs1_efuse.h>
+#include <status.h>
+
+void security_load_revoke_bits(void)
+{
+	uint32_t key_revoke = boot_rom_global.config_fuse & EFUSE_SYSCFG_KEY_REVOCATION_BITS;
+
+	if (boot_rom_global.fake_fuse_en)
+		boot_rom_status_update(PROGRESS_REVOKE_LOC_SPI_FLASH);
+	else
+		boot_rom_status_update(PROGRESS_REVOKE_LOC_ASIC_EFUSE);
+
+	/* store key_revoke map in PSOC */
+	regmap_writel(key_revoke, IO0_H0_PSOC_BOOT_SECURITY(mmPSOC_SECURITY_KEY_REVOKE));
+
+	boot_rom_global.sec_data.key_revoke = key_revoke;
+}
diff --git a/arch/fs1/arc/dev_security.h b/arch/fs1/arc/dev_security.h
new file mode 100644
index 0000000..ecf5155
--- /dev/null
+++ b/arch/fs1/arc/dev_security.h
@@ -0,0 +1,13 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2024 HabanaLabs Ltd.
+ */
+
+#ifndef _DEV_SECURITY_H_
+#define _DEV_SECURITY_H_
+
+#define NUM_KEYS_IN_ROM					5
+
+void security_load_revoke_bits(void);
+
+#endif /* _DEV_SECURITY_H_ */
diff --git a/arch/fs1/arc/progress_bits.h b/arch/fs1/arc/progress_bits.h
index 8740f62..e41f964 100644
--- a/arch/fs1/arc/progress_bits.h
+++ b/arch/fs1/arc/progress_bits.h
@@ -29,110 +29,109 @@
 	PROGRESS_INIT_EFUSE_INIT_FAIL = 19,
 	PROGRESS_INIT_EFUSE_SPI = 20,
 	PROGRESS_INIT_EFUSE_SPI_SLV_TIMEDOUT = 21,
-	PROGRESS_REVOKE_LOC = 22,
-	PROGRESS_REVOKE_LOC_0 = 23,
-	PROGRESS_REVOKE_LOC_1 = 24,
-	PROGRESS_KEY_LOC = 25,
-	PROGRESS_KEY_LOC_0 = 26,
-	PROGRESS_KEY_LOC_1 = 27,
-	PROGRESS_VER_LOC = 28,
-	PROGRESS_VER_LOC_0 = 29,
-	PROGRESS_VER_LOC_1 = 30,
-	PROGRESS_SLV_DIE_WAITING_ON_RECOVERY_IND = 31,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_START = 32,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_VERSION_FAIL = 33,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_REVOKED = 34,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 35,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_AUTH_FAIL = 36,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_SUCCESS = 37,
-	PROGRESS_BRINGUP_PCI_PRI = 38,
-	PROGRESS_BRINGUP_PCI_PRI_LOADADDR_INVALID = 39,
-	PROGRESS_BRINGUP_PCI_PRI_STARTADDR_INVALID = 40,
-	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 41,
-	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 42,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 43,
-	PROGRESS_BRINGUP_PCI_PRI_INVALID_IMG_SIZE = 44,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 45,
-	PROGRESS_BRINGUP_PCI_PRI_INVALID_FLASH_OFFSET = 46,
-	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 47,
-	PROGRESS_BRINGUP_PCI_PRI_DMA = 48,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 49,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 50,
-	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 51,
-	PROGRESS_BRINGUP_PCI_SEC = 52,
-	PROGRESS_BRINGUP_PCI_SEC_LOADADDR_INVALID = 53,
-	PROGRESS_BRINGUP_PCI_SEC_STARTADDR_INVALID = 54,
-	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 55,
-	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 56,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 57,
-	PROGRESS_BRINGUP_PCI_SEC_INVALID_IMG_SIZE = 58,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 59,
-	PROGRESS_BRINGUP_PCI_SEC_INVALID_FLASH_OFFSET = 60,
-	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 61,
-	PROGRESS_BRINGUP_PCI_SEC_DMA = 62,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 63,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 64,
-	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 65,
-	PROGRESS_BOOT_ROM_TPM = 66,
-	PROGRESS_BOOT_ROM_TPM_INIT_FAIL = 67,
-	PROGRESS_BOOT_ROM_TPM_STARTUP_FAIL = 68,
-	PROGRESS_BOOT_ROM_TPM_PCR_EXTEND_FAIL = 69,
-	PROGRESS_BOOT_ROM_TPM_FINISH = 70,
-	PROGRESS_PPBOOT_START = 71,
-	PROGRESS_PPBOOT_FINISH = 72,
-	PROGRESS_BOOT_ROM_IN_RECOVERY = 73,
-	PROGRESS_BOOT_ROM_IN_ROM_RECOVERY = 74,
-	PROGRESS_BOOT_ROM_PCIE_LINK_START = 75,
-	PROGRESS_BOOT_ROM_PCIE_LINK_INIT = 76,
-	PROGRESS_BOOT_ROM_PCIE_PHY_SRAM_INIT_TIMEDOUT = 77,
-	PROGRESS_BOOT_ROM_PCIE_PHY_PARA_STS_CLR_TIMEDOUT = 78,
-	PROGRESS_BOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 79,
-	PROGRESS_BOOT_ROM_PCIE_LINK_IATU_CONFIG = 80,
-	PROGRESS_BOOT_ROM_PCIE_LINK_FINISH = 81,
-	PROGRESS_BOOT_ROM_D2D_LINK_START = 82,
-	PROGRESS_BOOT_ROM_D2D_LINK_STS_BUSY = 83,
-	PROGRESS_BOOT_ROM_D2D_LINK_INTER_DIE_SYNC = 84,
-	PROGRESS_BOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 85,
-	PROGRESS_BOOT_ROM_D2D_LINK_INIT_SUCCESS = 86,
-	PROGRESS_BOOT_ROM_RECOVERY_END = 87,
-	PROGRESS_BOOT_ROM_IN_RECOVERY_IMG = 88,
-	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_EXCEEDS_MAX_SIZE = 89,
-	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_DMA_START = 90,
-	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_CRC_FAIL = 91,
-	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_UNALIGNED_DEST = 92,
-	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_INVALID_IMG_SIZE = 93,
-	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_UNALIGNED_IMG_SIZE = 94,
-	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_INVALID_FLASH_OFFSET = 95,
-	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_DMA_ERROR = 96,
-	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_DMA_FINISH = 97,
-	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_HDR_MAGIC_FAIL = 98,
-	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_INVALID_IMG_OFFSET = 99,
-	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_VERIFICATION_FAIL = 100,
-	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_FAIL = 101,
-	PROGRESS_RECOVERY_IMG_BOOT_DESC_VERIFY_START = 102,
-	PROGRESS_RECOVERY_IMG_BOOT_DESC_VERIFY_VERSION_FAIL = 103,
-	PROGRESS_RECOVERY_IMG_BOOT_DESC_VERIFY_KEY_REVOKED = 104,
-	PROGRESS_RECOVERY_IMG_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 105,
-	PROGRESS_RECOVERY_IMG_BOOT_DESC_VERIFY_AUTH_FAIL = 106,
-	PROGRESS_RECOVERY_IMG_BOOT_DESC_VERIFY_SUCCESS = 107,
-	PROGRESS_RECOVERY_IMG_IMG0 = 108,
-	PROGRESS_RECOVERY_IMG_IMG0_LOADADDR_INVALID = 109,
-	PROGRESS_BOOT_REOCVERY_IMG_IMG0_STARTADDR_INVALID = 110,
-	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_START = 111,
-	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_CRC_FAILED = 112,
-	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_UNALIGNED_DEST = 113,
-	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_INVALID_IMG_SIZE = 114,
-	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_UNALIGNED_IMG_SIZE = 115,
-	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_INVALID_FLASH_OFFSET = 116,
-	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_ERROR = 117,
-	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_FINISH = 118,
-	PROGRESS_BOOT_REOCVERY_IMG_IMG0_VERIFICATION_FAIL = 119,
-	PROGRESS_BOOT_REOCVERY_IMG_IMG0_START = 120,
-	PROGRESS_BOOT_REOCVERY_IMG_IMG0_FINISH = 121,
-	PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_START = 122,
-	PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_FINISH = 123,
-	PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_START = 124,
-	PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_FINISH = 125,
+	PROGRESS_REVOKE_LOC_ASIC_EFUSE = 22,
+	PROGRESS_REVOKE_LOC_SPI_FLASH = 23,
+	PROGRESS_KEY_LOC = 24,
+	PROGRESS_KEY_LOC_0 = 25,
+	PROGRESS_KEY_LOC_1 = 26,
+	PROGRESS_VER_LOC = 27,
+	PROGRESS_VER_LOC_0 = 28,
+	PROGRESS_VER_LOC_1 = 29,
+	PROGRESS_SLV_DIE_WAITING_ON_RECOVERY_IND = 30,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_START = 31,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_VERSION_FAIL = 32,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_REVOKED = 33,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 34,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_AUTH_FAIL = 35,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_SUCCESS = 36,
+	PROGRESS_BRINGUP_PCI_PRI = 37,
+	PROGRESS_BRINGUP_PCI_PRI_LOADADDR_INVALID = 38,
+	PROGRESS_BRINGUP_PCI_PRI_STARTADDR_INVALID = 39,
+	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 40,
+	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 41,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 42,
+	PROGRESS_BRINGUP_PCI_PRI_INVALID_IMG_SIZE = 43,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 44,
+	PROGRESS_BRINGUP_PCI_PRI_INVALID_FLASH_OFFSET = 45,
+	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 46,
+	PROGRESS_BRINGUP_PCI_PRI_DMA = 47,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 48,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 49,
+	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 50,
+	PROGRESS_BRINGUP_PCI_SEC = 51,
+	PROGRESS_BRINGUP_PCI_SEC_LOADADDR_INVALID = 52,
+	PROGRESS_BRINGUP_PCI_SEC_STARTADDR_INVALID = 53,
+	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 54,
+	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 55,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 56,
+	PROGRESS_BRINGUP_PCI_SEC_INVALID_IMG_SIZE = 57,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 58,
+	PROGRESS_BRINGUP_PCI_SEC_INVALID_FLASH_OFFSET = 59,
+	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 60,
+	PROGRESS_BRINGUP_PCI_SEC_DMA = 61,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 62,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 63,
+	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 64,
+	PROGRESS_BOOT_ROM_TPM = 65,
+	PROGRESS_BOOT_ROM_TPM_INIT_FAIL = 66,
+	PROGRESS_BOOT_ROM_TPM_STARTUP_FAIL = 67,
+	PROGRESS_BOOT_ROM_TPM_PCR_EXTEND_FAIL = 68,
+	PROGRESS_BOOT_ROM_TPM_FINISH = 69,
+	PROGRESS_PPBOOT_START = 70,
+	PROGRESS_PPBOOT_FINISH = 71,
+	PROGRESS_BOOT_ROM_IN_RECOVERY = 72,
+	PROGRESS_BOOT_ROM_IN_ROM_RECOVERY = 73,
+	PROGRESS_BOOT_ROM_PCIE_LINK_START = 74,
+	PROGRESS_BOOT_ROM_PCIE_LINK_INIT = 75,
+	PROGRESS_BOOT_ROM_PCIE_PHY_SRAM_INIT_TIMEDOUT = 76,
+	PROGRESS_BOOT_ROM_PCIE_PHY_PARA_STS_CLR_TIMEDOUT = 77,
+	PROGRESS_BOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 78,
+	PROGRESS_BOOT_ROM_PCIE_LINK_IATU_CONFIG = 79,
+	PROGRESS_BOOT_ROM_PCIE_LINK_FINISH = 80,
+	PROGRESS_BOOT_ROM_D2D_LINK_START = 81,
+	PROGRESS_BOOT_ROM_D2D_LINK_STS_BUSY = 82,
+	PROGRESS_BOOT_ROM_D2D_LINK_INTER_DIE_SYNC = 83,
+	PROGRESS_BOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 84,
+	PROGRESS_BOOT_ROM_D2D_LINK_INIT_SUCCESS = 85,
+	PROGRESS_BOOT_ROM_RECOVERY_END = 86,
+	PROGRESS_BOOT_ROM_IN_RECOVERY_IMG = 87,
+	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_EXCEEDS_MAX_SIZE = 88,
+	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_DMA_START = 89,
+	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_CRC_FAIL = 90,
+	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_UNALIGNED_DEST = 91,
+	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_INVALID_IMG_SIZE = 92,
+	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_UNALIGNED_IMG_SIZE = 93,
+	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_INVALID_FLASH_OFFSET = 94,
+	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_DMA_ERROR = 95,
+	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_DMA_FINISH = 96,
+	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_HDR_MAGIC_FAIL = 97,
+	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_INVALID_IMG_OFFSET = 98,
+	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_VERIFICATION_FAIL = 99,
+	PROGRESS_INIT_RECOVERY_IMG_BOOT_DESC_FAIL = 100,
+	PROGRESS_RECOVERY_IMG_BOOT_DESC_VERIFY_START = 101,
+	PROGRESS_RECOVERY_IMG_BOOT_DESC_VERIFY_VERSION_FAIL = 102,
+	PROGRESS_RECOVERY_IMG_BOOT_DESC_VERIFY_KEY_REVOKED = 103,
+	PROGRESS_RECOVERY_IMG_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 104,
+	PROGRESS_RECOVERY_IMG_BOOT_DESC_VERIFY_AUTH_FAIL = 105,
+	PROGRESS_RECOVERY_IMG_BOOT_DESC_VERIFY_SUCCESS = 106,
+	PROGRESS_RECOVERY_IMG_IMG0 = 107,
+	PROGRESS_RECOVERY_IMG_IMG0_LOADADDR_INVALID = 108,
+	PROGRESS_BOOT_REOCVERY_IMG_IMG0_STARTADDR_INVALID = 109,
+	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_START = 110,
+	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_CRC_FAILED = 111,
+	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_UNALIGNED_DEST = 112,
+	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_INVALID_IMG_SIZE = 113,
+	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_UNALIGNED_IMG_SIZE = 114,
+	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_INVALID_FLASH_OFFSET = 115,
+	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_ERROR = 116,
+	PROGRESS_BOOT_REOCVERY_IMG_IMG0_DMA_FINISH = 117,
+	PROGRESS_BOOT_REOCVERY_IMG_IMG0_VERIFICATION_FAIL = 118,
+	PROGRESS_BOOT_REOCVERY_IMG_IMG0_START = 119,
+	PROGRESS_BOOT_REOCVERY_IMG_IMG0_FINISH = 120,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_START = 121,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_FINISH = 122,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_START = 123,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_FINISH = 124,
 };
 
 #endif /* _PROGRESS_BITS_H_ */
diff --git a/common/security.c b/common/security.c
index 912b05e..c6ef7f0 100644
--- a/common/security.c
+++ b/common/security.c
@@ -6,6 +6,24 @@
  */
 
 #include <boot_rom.h>
+#include <dev_security.h>
+
+int security_check_key_revoke_status(uint32_t key_index)
+{
+	uint8_t key_revoke;
+
+	if (key_index >= NUM_KEYS_IN_ROM)
+		return -1;
+
+	key_revoke = boot_rom_global.sec_data.key_revoke;
+
+	/* if the key is not revoked, return 1 */
+	if ((key_revoke & (1 << key_index)) == 0)
+		return 1;
+
+	/* no valid key found */
+	return -1;
+}
 
 static void security_handle_jtag_if(void)
 {
@@ -27,5 +45,5 @@
 		return;
 
 	security_load_sec_efuse_dword();
+	security_load_revoke_bits();
 }
-
diff --git a/common/security.h b/common/security.h
index cfdf06a..7c79fe4 100644
--- a/common/security.h
+++ b/common/security.h
@@ -7,5 +7,6 @@
 #define _SECURITY_H_
 
 void boot_rom_security_init(void);
+int security_check_key_revoke_status(uint32_t key_index);
 
 #endif /* _SECURITY_H_ */
